### PR TITLE
Fix/prov payments

### PR DESCRIPTION
**Pull request by TheLeoB**
## Description (optional)
Fixes the issue where inputs for non-custodial sell/swap/interest weren't updating in real time, as the amount was being changed on the form. This resulted in fee estimates for BTC send sometimes being a lot lower than fees used during the transaction.

## Testing Steps (optional)
Detail the steps required for the reviewer(s) to verify and test these changes.

### CODE REVIEW

File: packages/blockchain-wallet-v4-frontend/src/data/components/simpleBuy/sagas.ts [hunk 1]
```diff
@@ -939,7 +939,12 @@ export default ({
         Number(cryptoAmt),
         account.coin
       )
-      payment = yield payment.amount(paymentAmount)
+      if (payment.coin === 'BTC' || payment.coin === 'BCH') {
```
Comment (pedroapfilho) [inline packages/blockchain-wallet-v4-frontend/src/data/components/simpleBuy/sagas.ts:942]: Is this a good idea?

Comment (schnogz) [reply packages/blockchain-wallet-v4-frontend/src/data/components/simpleBuy/sagas.ts:942]: @plondon did you have a look at this?  any other ideas?

Comment (TheLeoB) [reply packages/blockchain-wallet-v4-frontend/src/data/components/simpleBuy/sagas.ts:942]: I changed this bit of code here to include all coins, (removing the BTC and BCH check), which fixed an issue for ETH but created another issue for XLM. When trying to send, I would see an error about reserve currency being greater than balance. 

https://github.com/blockchain/blockchain-wallet-v4-frontend/blob/86fbecd18239fdc28b7598c11f14ab5317da19aa/packages/blockchain-wallet-v4-frontend/src/data/components/swap/sagas.ts#L132

**Review by plondon — CHANGES_REQUESTED** [review]
Before diving deeper, adding an initial stamp of 👎

**Review by totongcute — COMMENTED** [review]
hi

**Pull request closed without merging**